tracks/elm/exercises/leap/mentoring.md

```diff
@@ -0,0 +1,90 @@
+This is a good exercise to discuss the describability of logic, as most students solve the problem using a conditional statement that is hard to describe in language, and there are other options.
+
+This exercise introduces `Basics.remainderBy`, the `let` / `in` syntax and partial application / currying.
+
+### Reasonable solutions
+
+The easiest / most common solution probably involves partially applying `Basics.remainderBy` and using a slightly complicated if statement:
+
+```elm
+isLeapYear : Int -> Bool
+isLeapYear year =
+    let
+        divisibleBy number = 
+            Basics.remainderBy number year == 0         
+    in
+        (
+            divisibleBy 4 && 
+            not (divisibleBy 100)
+        )
+        || divisibleBy 400
+```
+
+The complicated if statement is a bit of a problem, as it is hard to describe in language, and hence talking about the code with others is difficult.
+
+Using a case statement turns the if statement in to a truth table, and is probably a bit clearer, but at the expense of some extra code:
```

Comment:
The following snippet is one that I give as feedback on the Haskell track:

> I would categorize your solution strategy as "truth-table-based". I would also argue that such solutions are generally not ideal, because the chunks of `True`s and `False`s can be hard to mentally map back, so you have to move your eyes forth and back between the table and the "table header".
>
> The syntactic alternative closest to truth tables that is easier to read is use of [guards](http://learnyouahaskell.com/syntax-in-functions#guards-guards):
>
> ```haskell
> isLeapYear :: Integer -> Bool
> isLeapYear year
>   | isDivisibleBy 400 = True
>   | isDivisibleBy 100 = False
>   | isDivisibleBy   4 = True
>   | otherwise         = False
>    where
>      isDivisibleBy n = year `rem` n == 0
> ```

So I think this point can at least be debated.

Reply:
Hi Simon. Elm doesn't support the guard syntax, so that isn't a possibility.

I would say that the "truth-table" is clearer, easier to describe in words and takes less mental mapping effort than the conditional statement. It isn't my favourite solution though.

However, even though I think the mental mapping is easy, there is quite a lot of it to do, and you make a good point about moving your eyes back and forth.

I have incorporated your comments.